.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,54 @@
+name: Pull Request Validation
+
+on:
+  pull_request:
+    branches:
+      - trunk
+    types:
+      - "labeled"
+      - "unlabeled"
+      - "opened"
+      - "reopened"
+      - "synchronize"
+
+jobs:
+  check-labels:
+    name: Check Labels and Milestone
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check Type Labels
+        if: |
+          !contains( github.event.pull_request.labels.*.name, '[Type] Bug' ) &&
+          (
+            !contains( github.event.pull_request.labels.*.name, '[Type] Feature' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Type] Documentation' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Type] Discussion' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Type] Enhancement' )
+          )
+        uses: actions/github-script@v5
+        with:
+          script: |
+            core.setFailed( 'Pull request must have a [Type] label.' )
+
+      - name: Check Focus or Infrastructure Labels
+        if: |
+          !contains( github.event.pull_request.labels.*.name, '[Focus] Images' ) &&
+          (
+            !contains( github.event.pull_request.labels.*.name, '[Focus] JavaScript' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Focus] Measurement' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Focus] Object caching' ) &&
+            !contains( github.event.pull_request.labels.*.name, '[Focus] Site Health' ) &&
+            !contains( github.event.pull_request.labels.*.name, 'Infrastructure' )
+          )
```

Comment:
Similar to my comment above: I think this could be simplified and made easier to maintain by checking the following:
* the PR does not have any label starting with `[Focus] ...`
* _and_ the PR also does not have the `Infrastructure` label

That way we don't have to update this code whenever we add a new focus.